src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -375,4 +380,79 @@ absl::StatusOr<std::string> AkidFromCrl(X509_CRL* crl) {
   return ret;
 }
 
+bool IsPemCertificateChainNonEmptyAndValid(absl::string_view cert_chain_pem) {
+  if (cert_chain_pem.empty()) {
+    return false;
+  }
+  BIO* in = BIO_new_mem_buf(cert_chain_pem.data(), cert_chain_pem.size());
+  if (in == nullptr) {
+    return false;
+  }
+  uint8_t* cert_data = nullptr;
+  long cert_length = 0;
+  size_t number_of_valid_certs = 0;
+  while (PEM_bytes_read_bio(&cert_data, &cert_length, /*pnm=*/nullptr,
+                            PEM_STRING_X509, in, /*cb=*/nullptr,
```

Comment:
Any reason this isn't just using `PEM_read_bio_X509`?

Reply:
Thanks for the suggestion - I didn't know about this API! Switched to use it.